test/lint/lint-files.sh

```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env bash
+
+export LC_ALL=C
+
+set -e
+cd "$(dirname $0)/../.."
+test/lint/lint-files.py
```

Comment:
Just a question (not a blocker): why does this need a bash wrapper?

Reply:
I believe the main reason is the the `lint-all.sh` script that CI runs is currently setup to automatically run all `.sh` tests in the `test/lint` directory, but won't automatically run `.py` files.

In addition it allows lint tests [to opt in or out of locale dependence](https://github.com/bitcoin/bitcoin/blob/7fcf53f7b4524572d1d0c9a5fdc388e87eb02416/test/lint/lint-all.sh#L10-L12)

If we start to move more towards using python based linters, it probably makes sense to make some updates here, so we can get rid of the  requirement for the `.sh` wrapper.

@practicalswift may have more context here.